File: CON BDD/EUREKABANK_RESTFULL_JAVA_GR06/08. SERVIDOR/eurekabank_restful_java_gr06/src/main/java/ec/edu/monster/modelo/Resultado.java
package ec.edu.monster.modelo;

/**
 * DTO de respuesta para operaciones bancarias. En lugar de devolver solo un
 * boolean, el servicio informa exito/fallo, un mensaje legible y, cuando
 * aplica, el saldo resultante.
 */
public class Resultado {

    private boolean exito;
    private String mensaje;
    private double saldo;

    public Resultado() {
    }

    public Resultado(boolean exito, String mensaje) {
        this.exito = exito;
        this.mensaje = mensaje;
    }

    public Resultado(boolean exito, String mensaje, double saldo) {
        this.exito = exito;
        this.mensaje = mensaje;
        this.saldo = saldo;
    }

    public static Resultado ok(String mensaje, double saldo) {
        return new Resultado(true, mensaje, saldo);
    }

    public static Resultado error(String mensaje) {
        return new Resultado(false, mensaje);
    }

    public boolean isExito() { return exito; }
    public void setExito(boolean exito) { this.exito = exito; }

    public String getMensaje() { return mensaje; }
    public void setMensaje(String mensaje) { this.mensaje = mensaje; }

    public double getSaldo() { return saldo; }
    public void setSaldo(double saldo) { this.saldo = saldo; }
}

File: CON BDD/EUREKABANK_REST_DOTNET_GR06/08. SERVIDOR/Models/Resultado.cs
namespace SERVIDOR.Models;

public class Resultado
{
    public bool Exitoso { get; set; }
    public string Mensaje { get; set; } = string.Empty;
    public double Saldo { get; set; }

    public Resultado() { }

    public Resultado(bool exitoso, string mensaje)
    {
        Exitoso = exitoso;
        Mensaje = mensaje;
    }

    public Resultado(bool exitoso, string mensaje, double saldo)
    {
        Exitoso = exitoso;
        Mensaje = mensaje;
        Saldo = saldo;
    }

    public static Resultado Ok(string mensaje, double saldo) =>
        new Resultado(true, mensaje, saldo);

    public static Resultado Error(string mensaje) =>
        new Resultado(false, mensaje);
}
